File: Website/DesktopModules/Admin/ModuleCreator/Templates/C#/DAL2/templateController.cs
#region Copyright

// 
// Copyright (c) [YEAR]
// by [OWNER]
// 

#endregion

#region Using Statements

using System.Collections.Generic;
using DotNetNuke.Data;

#endregion

namespace [OWNER].[MODULE]
{
    public class [CONTROL]Controller
    {
        public void Add[CONTROL]([CONTROL]Info [CONTROL])
        {
            using (IDataContext ctx = DataContext.Instance())
            {
                var rep = ctx.GetRepository<[CONTROL]Info>();
                rep.Insert([CONTROL]);
            }
        }

        public void Delete[CONTROL]([CONTROL]Info [CONTROL])
        {
            using (IDataContext ctx = DataContext.Instance())
            {
                var rep = ctx.GetRepository<[CONTROL]Info>();
                rep.Delete([CONTROL]);
            }
        }

        public IEnumerable<[CONTROL]Info> Get[CONTROL]s(int moduleId)
        {
            IEnumerable<[CONTROL]Info> [CONTROL]s;

            using (IDataContext ctx = DataContext.Instance())
            {
                var rep = ctx.GetRepository<[CONTROL]Info>();
                [CONTROL]s = rep.Get(moduleId);
            }
            return [CONTROL]s;
        }

        public [CONTROL]Info Get[CONTROL](int [CONTROL]Id, int moduleId)
        {
            [CONTROL]Info [CONTROL];

            using (IDataContext ctx = DataContext.Instance())
            {
                var rep = ctx.GetRepository<[CONTROL]Info>();
                [CONTROL] = rep.GetById([CONTROL]Id, moduleId);
            }
            return [CONTROL];
        }

        public void Update[CONTROL]([CONTROL]Info [CONTROL])
        {
            using (IDataContext ctx = DataContext.Instance())
            {
                var rep = ctx.GetRepository<[CONTROL]Info>();
                rep.Update([CONTROL]);
            }
        }
    }
}

File: Website/DesktopModules/Admin/ModuleCreator/Templates/VB/DAL2/templateController.vb
#Region "Copyright"

' 
' Copyright (c) _YEAR_
' by _OWNER_
' 

#End Region

#Region "Using Statements"

Imports System.Collections.Generic
Imports DotNetNuke.Data

#End Region

Namespace _OWNER_._MODULE_
    Public Class _MODULE_Controller
        Public Sub Add_MODULE_(_MODULE_ As _MODULE_Info)
            Using ctx As IDataContext = DataContext.Instance()
                Dim rep = ctx.GetRepository(Of _MODULE_Info)()
                rep.Insert(_MODULE_)
            End Using
        End Sub

        Public Sub Delete_MODULE_(_MODULE_ As _MODULE_Info)
            Using ctx As IDataContext = DataContext.Instance()
                Dim rep = ctx.GetRepository(Of _MODULE_Info)()
                rep.Delete(_MODULE_)
            End Using
        End Sub

        Public Function Get_MODULE_s(moduleId As Integer) As IEnumerable(Of _MODULE_Info)
            Dim _MODULE_s As IEnumerable(Of _MODULE_Info)

            Using ctx As IDataContext = DataContext.Instance()
                Dim rep = ctx.GetRepository(Of _MODULE_Info)()
                _MODULE_s = rep.[Get](moduleId)
            End Using
            Return _MODULE_s
        End Function

        Public Function Get_MODULE_(_MODULE_Id As Integer, moduleId As Integer) As _MODULE_Info
            Dim _MODULE_ As _MODULE_Info

            Using ctx As IDataContext = DataContext.Instance()
                Dim rep = ctx.GetRepository(Of _MODULE_Info)()
                _MODULE_ = rep.GetById(_MODULE_Id, moduleId)
            End Using
            Return _MODULE_
        End Function

        Public Sub Update_MODULE_(_MODULE_ As _MODULE_Info)
            Using ctx As IDataContext = DataContext.Instance()
                Dim rep = ctx.GetRepository(Of _MODULE_Info)()
                rep.Update(_MODULE_)
            End Using
        End Sub
    End Class
End Namespace
